content/articles/django-custom-template-tags-and-filters/index.md

```diff
@@ -0,0 +1,258 @@
+
+register = template.Library()
+
+
+@register.filter(name='upper')
+def upper(value):
+  return value.upper()
+@register.filter(name='modify_name')
+def modify_name(value):
+    return value.title()
+@register.filter(name='get_color')
+def color(value):
+    if value:
+        return "#FF0000"
+    
+@register.simple_tag(name='my_contacts')
+def my_contacts():
+    return Contact.objects.all().count()
+@register.simple_tag(name='current_date')
+def current_date(format):
+    return datetime.datetime.now().strftime(format)
+```
+
+Change the html to this:
+
+```html
+{% load custom_tags %}
+{% block content %}
+<h3>You have {% my_contacts %} contacts</h3>
+{% for contact in contacts %}
+<div class="flex-box-container">
+    <div style='background-color: {{contact.name|get_color}};'><p >{{contact.name|upper}}</p></div>
+    <div><b>Created:</b>  {% current_date "%d/%m/%Y" %} </div>
+</div>
+{% endfor %}
+
+{% endblock content %}
+```
+
+The resulting template would look as shown below:
+
+![Contact App](/engineering-education/Django-custom-template-tags-and-filters/image3.png)
+
+Likewise, you can create your own simple tags and use them how you deem fit.
+
+### Inclusion Tags. 
+This tags work by rendering another template and are essential when working with information that is to be found in several pages.
+
+Let’s create a simple inclusion tag that counts the number of users.
+
+Create a `users.html` file that will be rendered by the inclusion tag. In your `custom_tags.py` file, edit and append the following:
+
+```python
+from Django.contrib.auth.models import User
+@register.inclusion_tag('users.html')
+def show_users():
+      obj = User.objects.values_list('username', flat=True)
+      return {'users': obj}
+```
+
+The template file `users.html` will look as shown below:
+```html
+<ul>
+{% for user in users %}
+    <li> {{ user }} </li>
+{% endfor %}
+</ul>
+```
+
+Then make contact.html look like this:
+
+```html
+{% comment %} users.html {% endcomment %}
+{% load custom_tags %}
+
+
+{% block content %}
+<h3>You have {% my_contacts %} contacts</h3>
+{% for contact in contacts %}
+<div class="flex-box-container">
+    <div style='background-color: {{contact.name|get_color}};'><p >{{contact.name|upper}}</p></div>
+    <div><b>Created:</b>  {% current_date "%d/%m/%Y" %} </div>
+</div>
+{% endfor %}
+{% show_users %}
+{% endblock content %}
+
+```
+
+The resulting template will now look as shown below:
+
+![Contact App](/engineering-education/Django-custom-template-tags-and-filters/image4.png)
+
+Notice that by adding `show_users` tag, we now have a user added below the contacts.
+
+### Conclusion
+You have now successfully created your custom template tags and filters.You should now be able to create new template tags and filters and apply them anywhere they are needed in your Django application.
+
+I also recommend look through the Django documentation to have a clear understanding of how tags and filters work.
```

Comment:
Provide a link to the section in the Django docs that talks about tags and filters.

Reply:
Done